_includes/components/mermaid.html

```diff
@@ -1,5 +1,45 @@
+{% comment %}
+The complexity of this file comes from a breaking change in Mermaid v10; mermaid.init has been deprecated (and supposedly, didn't work earlier?).
+
+So, we use a rough compile-time heuristic to see if the user's Mermaid version is >= 10; if not, we fall back to the "old" init syntax.
+
+If a user is using a custom mermaid file and doesn't specify a version, we default to the < v10 behaviour. Users who use version v10 or above should specify this in the version key.
+{% endcomment %}
+
+{% if site.mermaid.version %}
+  {% assign mermaid_major_version = site.mermaid.version | split: "." | first | plus: 0 %}
+{% else %}
+  {% assign mermaid_major_version = 9 %}
+{% endif %}
+
+{% unless mermaid_major_version < 10 %}
```

Comment:
Using `unless` with an `else` part is unusual, and it's unclear whether it involves a double negation of the condition. I'd prefer to use `if-else`.